### ISSUE TITLE
Try to use the `fit_curve` and `predict_curve` processes

### ISSUE THREAD

**Issue by clausmichele** [interTwin-eu/openeo-flood-mapper-local#4]
In openEO we defined and implemented two processes to work with harmonic models fitting and predicting, which are:

`fit_curve`:
- definition https://processes.openeo.org/#fit_curve
- implementation: https://github.com/Open-EO/openeo-processes-dask/blob/c09c805e337f357af48e393cc84a0719639b6b05/openeo_processes_dask/process_implementations/ml/curve_fitting.py#L17

`predict_curve`:
- definition: https://processes.openeo.org/#predict_curve
- implementation: https://github.com/Open-EO/openeo-processes-dask/blob/c09c805e337f357af48e393cc84a0719639b6b05/openeo_processes_dask/process_implementations/ml/curve_fitting.py#L102

Currently in the code there is a prediction step for a single date (when it's already known there's a flood). I guess it would be interesting to predict values for a longer time series and then work with the result, implementing the flood detection/alarm part.

**Comment by SchwammDev** [comment]
Sure, don't see a downside here. I'd suggest adding this as a small refactoring step during our next update of the algorithm.

**Comment by fl0roth** [comment]
Thank you for pointing this out, @clausmichele! 
We should definitely test the prediction based on the `predict_curve` process, and I think we can foresee to allow the flood mapping for multiple time steps in the future. Just to clarify: The prediction can be made for any date, not only those for which we already know there is a flood.
Using the `fit_curve` process to retrieve the parameters on the fly would be an interesting stretch goal, which could be investigated after the workflow is fully set up.

**Comment by clausmichele** [comment]
Yes @fl0roth, I know that the prediction can be run on any date with the current code. However, if we would like to work with a flood prediction datacube with many time steps and do further processing, in my opinion using the `predict_curve` could help out.
